### PR TITLE
Fix LifecycleListener when SDK started after onCreate

### PR DESCRIPTION
**Pull request by russelarms**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

A few sentences describing the overall goals of the pull request.

## Related PRs or issues

[AB#79328](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/79328)

### CODE REVIEW

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 1]
```diff
@@ -235,4 +235,67 @@ public void activityTransitionTest() {
...
+        mApplicationLifecycleListener.onActivityPaused(mActivityMock);
+        verify(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
+
+        /* Call onActivityStopped. */
+        mApplicationLifecycleListener.onActivityStopped(mActivityMock);
+        verify(callbacks, times(2)).onApplicationEnterBackground();
+
+        /* Call onActivityStarted. */
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(callbacks, times(1)).onApplicationEnterForeground();
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java:299]:
```suggestion
        verify(callbacks).onApplicationEnterForeground();
```

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 2]
```diff
@@ -235,4 +235,67 @@ public void activityTransitionTest() {
...
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(callbacks1, times(1)).onApplicationEnterForeground();
+
+        /* Go to another activity. */
+        mApplicationLifecycleListener.onActivityPaused(mActivityMock);
+        mApplicationLifecycleListener.onActivityStarted(secondActivityMock);
+        mApplicationLifecycleListener.onActivityResumed(secondActivityMock);
+
+        /* Verify that onApplicationEnterForeground was called once. */
+        verify(callbacks1, times(1)).onApplicationEnterForeground();
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java:273]:
```suggestion
        verify(callbacks1).onApplicationEnterForeground();
```

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 3]
```diff
@@ -235,4 +235,67 @@ public void activityTransitionTest() {
...
+        mApplicationLifecycleListener.onActivityPaused(mActivityMock);
+        verify(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
+
+        /* Call onActivityStopped. */
+        mApplicationLifecycleListener.onActivityStopped(mActivityMock);
+        verify(callbacks1, times(2)).onApplicationEnterBackground();
+
+        /* Call onActivityStarted. */
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(callbacks1, times(1)).onApplicationEnterForeground();
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java:265]:
```suggestion
        verify(callbacks1).onApplicationEnterForeground();
```

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 4]
```diff
@@ -235,4 +235,67 @@ public void activityTransitionTest() {
...
         verifyNoMoreInteractions(callbacks1, callbacks2);
     }
+
+    @Test
+    public void skipCallOnStart() {
+
+        /* Prepare data. */
+        Activity secondActivityMock = mock(Activity.class);
+        Bundle mockBundle = mock(Bundle.class);
+        ApplicationLifecycleListener.ApplicationLifecycleCallbacks callbacks1 = mock(ApplicationLifecycleListener.ApplicationLifecycleCallbacks.class);
```
Comment (guperrot) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java:245]: There is no callbacks2

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 5]
```diff
@@ -235,4 +235,67 @@ public void activityTransitionTest() {
...
+    public void skipCallOnResume() {
+
+        /* Prepare data. */
+        Bundle mockBundle = mock(Bundle.class);
+        ApplicationLifecycleListener.ApplicationLifecycleCallbacks callbacks = mock(ApplicationLifecycleListener.ApplicationLifecycleCallbacks.class);
+        mApplicationLifecycleListener.registerApplicationLifecycleCallbacks(callbacks);
+
+        /* Skip call onActivityStarted and onActivityResume. */
+
+        /* Call onActivityResumed. */
```
Comment (guperrot) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java:286]: Wrong comment, also why are we using this callback in the test?

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 6]
```diff
@@ -49,7 +45,6 @@ public void setUp() {
 
             @Override
             public Void answer(InvocationOnMock invocation) {
-                ((Runnable) invocation.getArguments()[0]).run();
                 return null;
             }
         }).when(mHandlerMock).postDelayed(any(Runnable.class), anyLong());
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java:50]: No need for `doAnswer` for just no-op

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java [hunk 7]
```diff
@@ -224,15 +220,91 @@ public void activityTransitionTest() {
...
+        /* Runnable for onActivityPaused is called after the timeout. */
+        postDelayedRunnable.getValue().run();
+
+        /* Verify onApplicationEnterBackground called. */
+        verify(callbacks).onApplicationEnterBackground();
+
+        /* Call onActivityStarted. */
+        mApplicationLifecycleListener.onActivityStarted(mActivityMock);
+        verify(callbacks).onApplicationEnterForeground();
+        verify(callbacks).onApplicationEnterForeground();
```
Comment (MatkovIvan) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/ApplicationLifecycleListenerTest.java:300]: The same thing is verified twice

**Pull request merged**